pkg/ingestor/api/api.go

```diff
@@ -52,6 +56,52 @@ func NewIngestorAPI(cfg *config.KubehoundConfig, puller puller.DataPuller, notif
 	}
 }
 
+// RehydrateLatest is just a GRPC wrapper around the Ingest method from the API package
+func (g *IngestorAPI) RehydrateLatest(ctx context.Context) ([]*grpc.IngestedCluster, error) {
+	// first level key are cluster names
+	directories, errRet := g.puller.ListFiles(ctx, "", false)
+	if errRet != nil {
+		return nil, errRet
+	}
+
+	res := []*grpc.IngestedCluster{}
+
+	for _, dir := range directories {
+		clusterName := strings.TrimSuffix(dir.Key, "/")
+
+		dumpKeys, err := g.puller.ListFiles(ctx, clusterName, true)
+		if err != nil {
+			return nil, err
+		}
+
+		if l := len(dumpKeys); l > 0 {
+			// extracting the latest runID
+			sort.Slice(dumpKeys, func(a, b int) bool {
+				return dumpKeys[a].ModTime.Before(dumpKeys[b].ModTime)
+			})
+			ingestTime := dumpKeys[l-1].ModTime
+			prefix := fmt.Sprintf("%s/kubehound_%s_", clusterName, clusterName)
+			runID := strings.TrimPrefix(dumpKeys[l-1].Key, prefix)
+			runID = strings.TrimSuffix(runID, writer.TarWriterExtension)
```

Comment:
This sounds like a function `ExtractRunIDFromKey` that should be defined just next to the function creating the key 🙈

Reply:
added a new object with a bunch of unit tests everywhere.